### PR TITLE
Fix guild table init

### PR DESCRIPTION
**Pull request by akotadi**
# Description

Fix logic when initializing the guild table, we should wait to the bot to be ready to be able to access to its guilds.

# Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# How Has This Been Tested?

Local testing along with some unit tests added

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by GoberInfinity — APPROVED** [review]

File: otter_welcome_buddy/database/db_guild.py [hunk 1]
```diff
@@ -21,8 +21,15 @@ def get_guild(
     def insert_guild(
         guild_model: GuildModel,
         session: Session,
+        override: bool = False,
     ) -> GuildModel:
         """Static method to insert a guild record"""
+        if not override:
+            current_guild_model = DbGuild.get_guild(guild_id=guild_model.id, session=session)
+            if current_guild_model is None:
```
Comment (Andresrodart) [inline otter_welcome_buddy/database/db_guild.py:29]: Why do we use `is not` and for the override, do we use the __bool__ method?

File: otter_welcome_buddy/startup/database.py [hunk 2]
```diff
@@ -8,14 +8,16 @@
...
 
-async def init_database(bot: Bot) -> None:
+def init_guild_table(bot: Bot) -> None:
+    """Verify that all the guilds that the bot is part of are in the database"""
+    for guild in bot.guilds:
+        print(f"Initializing guild {guild.name} [{guild.id}]")
+        with session_scope() as session:
+            guild_model = GuildModel(id=guild.id)
+            DbGuild.insert_guild(guild_model=guild_model, session=session)
+
```
Comment (Andresrodart) [inline otter_welcome_buddy/startup/database.py:18]: Extra space?

File: otter_welcome_buddy/database/db_guild.py (hunk 1 above)
Comment (akotadi) [reply otter_welcome_buddy/database/db_guild.py:29]: Where do we use `is not`? I don't get the question

File: otter_welcome_buddy/startup/database.py (hunk 2 above)
Comment (akotadi) [reply otter_welcome_buddy/startup/database.py:18]: Nope, between functions I try to let two spaces to avoid more confusion with python indenting

**Comment by akotadi** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #35** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/35?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #36** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/36?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #30** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/30?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #37** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/37?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #38** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/38?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #39** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/39?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #40** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/40?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #41** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/41?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #42** <a href="https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/42?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Proyecto-Nutria/otter-welcome-buddy/35?utm_source=stack-comment).

**Comment by akotadi** [comment]
Abandon as is not required anymore

**Pull request closed without merging**